fms/utils/serialization.py

```diff
@@ -55,6 +56,47 @@ def list_sources(architecture: str):
     return list(__adapters[architecture].keys())
 
 
+def __fms_weights_preprocessing(orig_sd: Mapping) -> Mapping:
+    new_sd = {}
+    for name, param in orig_sd.items():
+        # logic to handle weights coming from an older unfused model
+        if "attn.query" in name or "attn.key" in name or "attn.value" in name:
+            weight_type = name.split(".")[-1]
+
+            unfused_weights = [
+                re.sub(
+                    rf"attn.(query|key|value).{weight_type}",
+                    f"attn.query.{weight_type}",
+                    name,
+                ),
+                re.sub(
+                    rf"attn.(query|key|value).{weight_type}",
+                    f"attn.key.{weight_type}",
+                    name,
+                ),
+                re.sub(
+                    rf"attn.(query|key|value).{weight_type}",
+                    f"attn.value.{weight_type}",
+                    name,
```

Comment:
is this code supposed to work with roberta as well? based on above it looks like roberta has `attention.*` instead of `attn.*`. We could also add other models with different prefix keys right?

Reply:
This code will work for all fms models. The code above for `attention` is the huggingface name for roberta attention as that is the huggingface to fms adapter conversion